lib/fs.js

```diff
@@ -1571,9 +1698,139 @@ fs.realpathSync = function realpathSync(path, options) {
   else if (typeof options !== 'object')
     throw new TypeError('"options" must be a string or an object');
   nullCheck(path);
-  return binding.realpath(pathModule._makeLong(path), options.encoding);
+  const resolvedPath = pathModule.resolve(path.toString('utf8'));
+  const uvResult = binding.realpath(pathModule._makeLong(resolvedPath),
+                                    options.encoding);
+  if (realpathJSNeeded(resolvedPath, uvResult))
+  {
+    const jsResult = realpathSyncJS(resolvedPath, realpathCache);
+    return convertRealpathJSResult(jsResult, options.encoding);
+  } else {
+    return uvResult;
+  }
 };
 
+function realpathJS(p, cache, cb) {
+  if (typeof cb !== 'function') {
+    cb = maybeCallback(cache);
+    cache = null;
+  }
+
+  // make p is absolute
+  p = pathModule.resolve(p);
+
+  if (cache && Object.prototype.hasOwnProperty.call(cache, p)) {
+    return process.nextTick(cb.bind(null, null, cache[p]));
```

Comment:
This is how it was before switch to uv realpath. I would rather leave it as it was before it was removed, at least for now.